src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java

```diff
@@ -356,8 +356,8 @@ protected void createStatefulSet(String appId, AppDeploymentRequest request, Map
 				.addToMatchLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).endSelector()
 				.withVolumeClaimTemplates(persistentVolumeClaimBuilder.build()).withServiceName(appId)
 				.withPodManagementPolicy("Parallel").withReplicas(replicas).withNewTemplate().withNewMetadata()
-				.withLabels(idMap).addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).endMetadata().withSpec(podSpec)
-				.endTemplate().build();
+				.withLabels(idMap).addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).addToLabels(getDeploymentLabels(request))
+				.endMetadata().withSpec(podSpec).endTemplate().build();
 
 		StatefulSet statefulSet = new StatefulSetBuilder().withNewMetadata().withName(appId).withLabels(idMap)
 				.addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).endMetadata().withSpec(spec).build();
```

Comment:
I think we also need to set the `label` on StatefulSet `metadata` here.

Reply:
Ok let me see, I did verify all the labels got transferred over to the individual plods. I can check again

Reply:
@ilayaperumalg  thanks for review, I added the labels in StatefulSet metadata, I also added verification for all Stateful set metadata, pod template and pod metadata. Please review